pybind11/style.rst

```diff
@@ -59,33 +58,40 @@ Modules and source files
 
 .. _style-guide-pybind11-module-naming:
 
-Wrappers for a C++ header file SHOULD go in a Python module with a lowercased version of the header file name
--------------------------------------------------------------------------------------------------------------
+Compiled for a package SHOULD go in a module and source file named by adding an underscore prefix to the package name.
```

Comment:
I think you want a subsection title before this.  "Compiled" doesn't make sense.  Maybe "Top-level wrapper code"?